client/src/test/java/io/pravega/client/segment/impl/SegmentOutputStreamTest.java

```diff
@@ -115,6 +118,28 @@ public void testConnectAndSend() throws SegmentSealedException, ConnectionFailed
         verifyNoMoreInteractions(connection);
     }
 
+    @Test(timeout = 10000)
+    public void testReconnectWorksWithTokenTaskInInternalExecutor() {
+        UUID cid = UUID.randomUUID();
+        PravegaNodeUri uri = new PravegaNodeUri("endpoint", SERVICE_PORT);
+
+        MockConnectionFactoryImpl cf = new MockConnectionFactoryImpl();
+        // create one thread on connection factory
+        cf.setExecutor(Executors.newSingleThreadScheduledExecutor());
+        CompletableFuture<Void> signal = new CompletableFuture<>();
+        MockControllerWithTokenTask controller = new MockControllerWithTokenTask(uri.getEndpoint(), uri.getPort(), cf,
+                true, signal);
+
+        ClientConnection connection = mock(ClientConnection.class);
+        cf.provideConnection(uri, connection);
+        SegmentOutputStreamImpl output = new SegmentOutputStreamImpl(SEGMENT, true, controller, cf, cid, segmentSealedCallback,
+                RETRY_SCHEDULE, DelegationTokenProviderFactory.create(controller, "scope", "stream"));
+        output.reconnect();
+
+        signal.join();
+        assertEquals(1, controller.countOfDelegationTokenCalls.get());
```

Comment:
you can use `mockito.verify(mock, times(k))` to count number of invocations.

Reply:
Sure, fixed. :)